pkg/api/pod/util.go

```diff
@@ -312,6 +312,10 @@ func DropDisabledVolumeMountsAlphaFields(volumeMounts []api.VolumeMount) {
 			volumeMounts[i].MountPropagation = nil
 		}
 	}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.VolumeSubpath) {
```

Comment:
This will cause kubelet to successfully mount but at the wrong path (volume root).  Are we expecting that the application container handles this appropriately by crashing?